### PR TITLE
bpo-30736: upgrade to Unicode 10.0

### PR DESCRIPTION
**Pull request by benjaminp**
Straightforward. While we're at it, though, strip trailing whitespace from generated tables.

### CODE REVIEW

File: Doc/whatsnew/3.7.rst [hunk 1]
```diff
@@ -237,6 +237,12 @@ xmlrpc.server
 its subclasses can be used as a decorator. (Contributed by Xiang Zhang in
 :issue:`7769`.)
 
+unicodedata
+-----------
+
+The internal database has been upgraded to use Unicode 10.0.0. (Contributed by
```
Comment (serhiy-storchaka) [inline Doc/whatsnew/3.7.rst]: Maybe add a link as in `whatsnew/3.6.rst`?

**Review by serhiy-storchaka — APPROVED** [review]
There are links to Unicode 8.0 in `Doc/reference/lexical_analysis.rst` and `Doc/library/stdtypes.rst`. Shouldn't they be updated too?

**Comment by serhiy-storchaka** [comment]
> While we're at it, though, strip trailing whitespace from generated tables.

Always wanted to do this.

**Comment by serhiy-storchaka** [comment]
As for the CI failure see [bpo-29887](https://bugs.python.org/issue29887).

**Comment by benjaminp** [comment]
Yes, I'm glad it fails now.

**Comment by srl295** [comment]
Rando here. Does this fetch directly from unicode.org? because https://github.com/dodo/node-unicodetable/issues/16#issuecomment-288306521

**Comment by benjaminp** [comment]
No, we shouldn't be.

**Pull request merged**